File: slack-poster/src/test/java/dev/number6/slackposter/dagger/TestSlackPosterComponent.java
package dev.number6.slackposter.dagger;

import dev.number6.slack.port.HttpPort;
import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {SlackPortModule.class,
        FakeSlackPosterConfigurationModule.class,
        FakeHttpModule.class})
@Singleton
public interface TestSlackPosterComponent extends SlackPosterComponent {

    HttpPort getRecordingHttpAdaptor();

}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/dagger/TestSlackPosterComponent.kt
package dev.number6.slackposter.dagger

import dagger.Component
import dev.number6.slackposter.port.SlackPosterPort
import javax.inject.Singleton

@Component(modules = [FakeSlackPosterConfigurationModule::class,
    RecordingSlackPosterModule::class])
@Singleton
interface TestSlackPosterComponent : SlackPosterComponent {
    fun recordingHttpAdaptor(): SlackPosterPort
}